### PR TITLE
build: update dependencies

### PR DESCRIPTION
**Pull request by havogt**
updates minimum gridtools_cpp version for #1699

### CODE REVIEW

File: src/gt4py/next/ffront/dialect_parser.py [hunk 1]
```diff
@@ -106,11 +106,11 @@ def get_location(self, node: ast.AST) -> SourceLocation:
 
         # `FixMissingLocations` ensures that all nodes have the location attributes
         assert hasattr(node, "lineno")
-        line = node.lineno + line_offset if node.lineno is not None else None
+        line = node.lineno + line_offset
```
Comment (havogt) [inline src/gt4py/next/ffront/dialect_parser.py:109]: let's see if that works (mypy complain because line, colum are not Optional)

Comment (tehrengruber) [reply src/gt4py/next/ffront/dialect_parser.py:109]: Ah great they fixed this in python.

File: src/gt4py/next/utils.py [hunk 2]
```diff
@@ -140,7 +140,7 @@ def impl(*args: Any | tuple[Any | tuple, ...]) -> _R | tuple[_R | tuple, ...]:
                 assert result_collection_constructor is not None
                 return result_collection_constructor(impl(*arg) for arg in zip(*args))
 
-            return fun(  # type: ignore[misc] # mypy not smart enough
+            return fun(  # type: ignore[call-arg, misc] # mypy not smart enough
```
Comment (tehrengruber) [inline src/gt4py/next/utils.py:143]: The cast is probably superfluous together with `call-arg`, but whatever.

**Review by tehrengruber — APPROVED** [review]

File: src/gt4py/next/utils.py (hunk 2 above)
Comment (havogt) [reply src/gt4py/next/utils.py:143]: I double checked. The cast actually fixes a `arg-type` error. The `call-arg` is for "too few arguments", not motivated to investigate further.

**Pull request merged**